Yousei.Connectors/Transmission/AddAction.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Threading.Tasks;
+using Transmission.API.RPC;
+using Yousei.Core;
+using Yousei.Shared;
+
+namespace Yousei.Connectors.Transmission
+{
+    internal class AddAction : FlowAction<AddArguments>
+    {
+        private readonly Client client;
+
+        public AddAction(Client client)
+        {
+            this.client = client;
+        }
+
+        protected override async Task Act(IFlowContext context, AddArguments arguments)
+        {
+            var url = await arguments.Url.Resolve<string>(context);
+            var downloadDirectory = await arguments.DownloadDirectory.Resolve<string>(context);
+
+            await context.SetData(
```

Comment:
Although this is technically correct I'd would first save the result in a variable before setting it in the context. This makes future debugging easier.